omod-2.0/src/test/java/org/openmrs/module/attachments/rest/AttachmentRestController2_0Test.java

```diff
@@ -94,6 +104,90 @@ public void shouldGetRefByUuid() {
 	public void shouldGetFullByUuid() {
 	}
 	
+	@Test
+	public void createAttachment_shouldCreateANewAttachment() throws Exception {
+		
+		SimpleObject attachment = new SimpleObject();
+		attachment.add("uuid", obs.getUuid());
+		attachment.add("comment", obs.getComment());
+		attachment.add("complexData", obs.getComplexData());
+		
+		String json = "{\"uuid\":\"" + getUuid() + "\",\"attachment\":\"" + attachment + "\"}";
+		
+		try {
+			json = new ObjectMapper().writeValueAsString(attachment);
+		}
+		catch (IOException e) {
+			log.error("attachment created", e);
+		}
+		
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI());
+		req.setContent(json.getBytes());
+		
+		SimpleObject result = null;
+		try {
+			result = deserialize(handle(req));
+		}
+		catch (Exception e) {
+			log.error("Attachment failed to be created!", e);
+		}
+		
+		Util.log("Created attachment", result);
+		
+		// Check existence in database
+		String uuid = (String) attachment.get("uuid");
+		Assert.assertNull(result);
+		String createdAttachment = obs.getUuid();
```

Comment:
@sherrif10   ,is this really the created attachment obj??

Reply:
Yeah i actually tested the created attachment below with ignorecase character